### PR TITLE
Multi vehicle DVL + acoustic comms test case

### PR DESCRIPTION
**Pull request by adityapande-1995**
This PR adds an integration test using 2 LRAUVs : Tethys and Daphne. Tethys is instructed to move at certain speed, its linear speed is observed using the DVL sensor, and sent to Daphne using acoustic comms. Daphne tries to catch up to Tethys with that speed.

### CODE REVIEW

File: lrauv_system_tests/test/vehicle/test_dvl_acoustic_comms.cc [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  node.Subscribe("/2/rx", acousticCbDaphne);
+
+  // Send move command to tethys and run the simulation.
+  for (int _ = 0; _ < 10; _++)
+  {
+    gz::msgs::Double cmdMsg;
+    cmdMsg.set_data(2);
+    cmdPubTethys.Publish(cmdMsg);
+    fixture.Step(10s);
+  }
```
Comment (hidmic) [inline lrauv_system_tests/test/vehicle/test_dvl_acoustic_comms.cc:111]: @adityapande-1995 meta: not necessary, but having Tethys describe a more complex trajectory (e.g. a circle) would be nice.

File: lrauv_system_tests/worlds/flat_seabed_two_vehicles.sdf [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+      <name>daphne</name>
+
+      <experimental:params>
+        <plugin element_id="gz::sim::systems::CommsEndpoint" action="modify">
+          <address>2</address>
+          <topic>2/rx</topic>
+        </plugin>
+        <plugin element_id="gz::sim::systems::Thruster" action="modify">
+          <namespace>daphne</namespace>
+          <joint_name>propeller_joint</joint_name>
```
Comment (hidmic) [inline lrauv_system_tests/worlds/flat_seabed_two_vehicles.sdf:368]: @adityapande-1995 hmm, why do we need to modify the thrusters?

**Review by hidmic — COMMENTED** [review]
First pass

File: lrauv_system_tests/worlds/flat_seabed_two_vehicles.sdf (hunk 2 above)
Comment (adityapande-1995) [reply lrauv_system_tests/worlds/flat_seabed_two_vehicles.sdf:368]: The namespace in ``tethys_equipped`` is hardcoded to ``tethys``, so by default all 3 LRAUVs listen for commands on ``/model/tethys/joint/propeller_joint/cmd_vel`` , I modified the namespace to ``daphne`` for the second one, so it listens for commands on ``/model/daphne/joint/propeller_joint/cmd_vel`` now.

File: lrauv_system_tests/test/vehicle/test_dvl_acoustic_comms.cc (hunk 1 above)
Comment (adityapande-1995) [reply lrauv_system_tests/test/vehicle/test_dvl_acoustic_comms.cc:132]: Added rudder angles and check for X coordinate : https://github.com/osrf/lrauv/pull/249/commits/f6b8e94cd1b734121295010e938094929b741d2b

File: lrauv_system_tests/test/vehicle/CMakeLists.txt [hunk 3]
```diff
@@ -34,6 +34,14 @@ target_link_libraries(test_dvl
     lrauv_gazebo_plugins::lrauv_gazebo_messages)
 gtest_discover_tests(test_dvl)
 
+add_executable(test_dvl_acoustic_comms test_dvl_acoustic_comms.cc)
+target_link_libraries(test_dvl_acoustic_comms
+  PUBLIC gtest_main
+  PRIVATE
+    ${PROJECT_NAME}_support
+    lrauv_gazebo_plugins::dvl_velocity_tracking)
```
Comment (hidmic) [inline lrauv_system_tests/test/vehicle/CMakeLists.txt:42]: @adityapande-1995 should be: 
```suggestion
    lrauv_gazebo_plugins::lrauv_gazebo_messages)
```

Comment (adityapande-1995) [reply lrauv_system_tests/test/vehicle/CMakeLists.txt]: Done

**Comment by hidmic** [comment]
@adityapande-1995 some tests are exceeding tolerances. This is new for non-mission tests. I wonder if something changed upstream.

**Comment by adityapande-1995** [comment]
> @adityapande-1995 some tests are exceeding tolerances and failing. This is new for non-mission tests. I wonder if something changed upstream.

Should I increase tolerances here ? @caguero

**Comment by adityapande-1995** [comment]
CI is passing now

**Comment by hidmic** [comment]
> CI is passing now

I'm a bit worried about that. Seems unstable.

**Comment by adityapande-1995** [comment]
I've opened a PR to increase tolerance. Lets see how that goes : https://github.com/osrf/lrauv/pull/255

**Pull request merged**